### PR TITLE
Re-enable remote tests

### LINKED ISSUE #2: enable IQM remote tests
currently disabled because of issues

### PR DESCRIPTION
**Pull request by cqc-alec**
Closes #2 .
A few updates to reflect changed API.
Also fix mypy script.

### CODE REVIEW

**Comment by cqc-alec** [comment]
No idea why the tests are failing on Linux and MacOS (and not running on Windows?). They pass for me lcoally.

**Comment by cqc-melf** [comment]
Are you sure you have set the env variable for the remote test locally?
I can try to run it and take a look, what do you think?

All extensions don't run the remote checks on windows, I would assume that it is an issue with the env variable on windows in python? Nothing I would worry about in this PR.

**Comment by cqc-alec** [comment]
> Are you sure you have set the env variable for the remote test locally?

Yup:
```
(env) alec@psi:~/r/pytket-iqm/tests$ PYTKET_RUN_REMOTE_TESTS=1 pytest
=============================================== test session starts ===============================================
platform linux -- Python 3.9.5, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/alec/r/pytket-iqm, configfile: pytest.ini
plugins: timeout-1.4.2, hypothesis-6.54.1, requests-mock-1.9.3
collected 10 items                                                                                                

backend_test.py .......                                                                                     [ 70%]
convert_test.py ...                                                                                         [100%]

=============================================== 10 passed in 10.88s ===============================================
```

**Comment by cqc-melf** [comment]
Maybe it is an issue with the credentials, how have you set them?
I will try this now.

**Comment by cqc-alec** [comment]
> Maybe it is an issue with the credentials, how have you set them?
I have them set in `~/.config/pytket/config.json`.

**Comment by cqc-melf** [comment]
Should work now

File: mypy-check [hunk 1]
```diff
@@ -4,15 +4,7 @@ set -evu
...
-MODULEDIR=$1
-
-# set MYPYPATH
-MYPYPATH="."
-for MOD in $(ls -d "$MODULEDIR"*); do
-    MOD_PATH="$(cd "$MOD" && pwd)"
-    MYPYPATH="$MYPYPATH:$MOD_PATH"
-done
-export MYPYPATH="$MYPYPATH"
+MYPYPATH=$1
```
Comment (cqc-melf) [inline mypy-check:7]: This is something we should change on the other repos, too?

File: docs/changelog.rst [hunk 2]
```diff
@@ -4,7 +4,7 @@ Changelog
 0.5.0 (unreleased)
 ------------------
 
-* Updated iqm-client version requirement to 4.2.
+* Updated iqm-client version requirement to 4.3.
```
Comment (cqc-melf) [inline docs/changelog.rst:7]: 0.4.0 has not been released yet, would suggest changing this

File: docs/changelog.rst [hunk 3]
```diff
@@ -4,7 +4,7 @@ Changelog
...
 
 0.4.0 (May 2022)
```
Comment (cqc-melf) [inline docs/changelog.rst:9]:
```suggestion
```

File: docs/changelog.rst [hunk 4]
```diff
@@ -4,7 +4,7 @@ Changelog
...
 ----------------
```
Comment (cqc-melf) [inline docs/changelog.rst:10]:
```suggestion
```

**Review by cqc-melf — COMMENTED** [review]
Some comments

File: mypy-check (hunk 1 above)
Comment (cqc-alec) [reply mypy-check:7]: Yes, probably.

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**